packages/waku/src/lib/utils/stream.ts

```diff
@@ -1,5 +1,25 @@
 // Utility functions for web streams (not Node.js streams)
 
+export const streamToString = async (
+  stream: ReadableStream,
+): Promise<string> => {
+  const decoder = new TextDecoder();
+  const reader = stream.getReader();
+  const outs: string[] = [];
+  let result: ReadableStreamReadResult<unknown>;
+  do {
+    result = await reader.read();
+    if (result.value) {
+      if (!(result.value instanceof Uint8Array)) {
+        throw new Error('Unexepected buffer type');
+      }
+      outs.push(decoder.decode(result.value, { stream: true }));
```

Comment:
Technically `TextDecoder` can break for arbitrary binary data encoded in RSC stream. One such exotic example is found in https://github.com/vitejs/vite-plugin-react/blob/2994900a697edcc5b251f7beed3104601780228d/packages/plugin-rsc/examples/basic/src/routes/payload/server.tsx#L11-L18. This passes `new TextEncoder().encode('🔥').reverse()` as client component props.

I think the alternative is either dump as binary file, or use base64 entirely, or mix two representation like https://github.com/devongovett/rsc-html-stream/blob/9b858445f4f5817470f373ae266dea04d5fcfac3/server.js#L74-L81

Reply:
Ah, thanks for the catch. Will use base64 instead.